src/validation.cpp

```diff
@@ -4357,7 +4398,7 @@ void CChainState::CheckBlockIndex(const Consensus::Params& consensusParams)
         if (!pindex->HaveTxsDownloaded()) assert(pindex->nSequenceId <= 0); // nSequenceId can't be set positive for blocks that aren't linked (negative is used for preciousblock)
         // VALID_TRANSACTIONS is equivalent to nTx > 0 for all nodes (whether or not pruning has occurred).
         // HAVE_DATA is only equivalent to nTx > 0 (or VALID_TRANSACTIONS) if no pruning has occurred.
-        if (!fHavePruned) {
+        if (!fHavePruned && (pindex->nHeight > snapshot_height)) {
```

Comment:
nit: "// Don't start being rigorous about missing block data if we've never pruned and we're within a utxo snapshot" ?